### PR TITLE
feature/capability config validation

### PR DESCRIPTION
**Pull request by HenryNguyen5**
- **Add workflow registration config validation for ocr3**
- **Remove unused config unmarshalling**
- **Add config validation to mercury trigger**

### CODE REVIEW

File: pkg/capabilities/consensus/ocr3/capability.go [hunk 1]
```diff
@@ -111,8 +111,14 @@ func newWorkflowConfig() *workflowConfig {
 }
 
 func (o *capability) RegisterToWorkflow(ctx context.Context, request capabilities.RegisterToWorkflowRequest) error {
+	err := o.ValidateConfig(request.Config)
```
Comment (DeividasK) [inline pkg/capabilities/consensus/ocr3/capability.go:114]:
```suggestion
	config, err := o.UnwrapAndValidateConfig(request.Config)
```

Comment (DeividasK) [reply pkg/capabilities/consensus/ocr3/capability.go:114]: I suspect that in most cases you'll want to unwrap before validating

File: pkg/capabilities/triggers/mercury_trigger_test.go [hunk 2]
```diff
@@ -281,6 +281,39 @@ func TestMultipleMercuryTriggers(t *testing.T) {
...
+	err = ts.ValidateConfig(config)
+	require.Error(t, err)
+
+	config = newConfigSingleFeed(t, "0x123zzz")
+	err = ts.ValidateConfig(config)
+	require.Error(t, err)
+
+	config = newConfigSingleFeed(t, "0x0001013ebd4ed3f5889FB5a8a52b42675c60c1a8c42bc79eaa72dcd922ac4292")
+	err = ts.ValidateConfig(config)
+	require.Error(t, err)
```
Comment (DeividasK) [inline pkg/capabilities/triggers/mercury_trigger_test.go:310]: It's not clear here what your expected error is. Could you use `require.ErrorIs` instead?

cc @cedric-cordenier @bolekk what's the expected best practice around this?

File: pkg/capabilities/consensus/ocr3/capability.go (hunk 1 above)
Comment (HenryNguyen5) [reply pkg/capabilities/consensus/ocr3/capability.go:114]: nice, having unwrap + validate in a single call would be great for DX

File: pkg/capabilities/triggers/mercury_trigger_test.go (hunk 2 above)
Comment (HenryNguyen5) [reply pkg/capabilities/triggers/mercury_trigger_test.go:310]: i copy / pasted the test cases from `FeedId.validate()` :see_no_evil:

**Comment by HenryNguyen5** [comment]
Closing out in favour of https://github.com/smartcontractkit/chainlink-common/pull/445

**Pull request closed without merging**